.github/workflows/check-change-note.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  pull_request_target:
+    types: [labeled, unlabeled, opened, synchronize, reopened, ready_for_review]
+    paths:
+      - "*/ql/src/**/*.ql"
+      - "*/ql/src/**/*.qll"
+      - "!**/experimental/**"
+
+jobs:
+  check-change-note:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if change note file is present
+        uses: dorny/paths-filter@7c0f15b688b020e95e00f15c61299b022f08ca95 # v2.8.0
+        id: paths_filter
+        with:
+          filters: |
+            change_note:
+              - '**/change-notes/*.md'
+      - name: Get PR labels
+        id: pr-labels
+        uses: joerick/pr-labels-action@0a4cc4ee0ab557ec0b1ae1157fa6fa7f9f4c494b # v1.0.6
```

Comment:
I think you can avoid this step entirely, because there is a `labels` object in the Actions `pull_request` event payload. Can you try dropping this step, and using `if: !contains(github.event.pull_request_target.labels, 'no-change-note-required')` as the condition of the step below?

Reply:
Oh, cool. I'll try that. I found conflicting information about whether `.labels` worked on the internet, and the above seemed to do the trick.

(And in that case, I might just go all the way and avoid external actions entirely. I was trying to embrace the `import left-pad` spirit when implementing this. :laughing:)

Reply:
I am generally of that mindset too: you can do quite a lot with just `gh api` to call the GitHub REST API (and it's already installed on the hosted runners).
My trick for working out what fields are in the event payload is simply to run workflows in a test repo, set an env var to hold `github.event`, and print it at the start. That said, the docs have recently improved: https://docs.github.com/en/actions/reference/events-that-trigger-workflows and https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads.

Reply:
Aha, got it! What I needed was `github.event.pull_request.labels.*.name`. Very nice.